junit-vintage-engine/src/test/java/org/junit/vintage/engine/DynamicTestsHelperRun.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.vintage.engine;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.engine.discovery.DiscoverySelectors;
+import org.junit.platform.launcher.Launcher;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestIdentifier;
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
+import org.junit.platform.launcher.core.LauncherFactory;
+
+class DynamicTestsHelperRun {
+
+	private Launcher launcher;
+	private Map<State, Integer> callCounts;
+	private LauncherDiscoveryRequestBuilder requestBuilder;
+	enum State {
+		failed, success,
+	}
+
+	@BeforeEach
+	void setUpLauncher() {
+		this.launcher = LauncherFactory.create();
+		this.callCounts = new HashMap<>();
+		this.requestBuilder = LauncherDiscoveryRequestBuilder.request();
```

Comment:
Please move the previous three initialization lines to the corresponding fields and make the fields `final`.